### ISSUE TITLE
Dashes in filenames replaced by underscores in 1.9

### ISSUE THREAD

**Issue by tttapa** [doxygen/doxygen#8377]
Doxygen 1.9 seems to replace the dashes in filenames by underscores. For example, I have a file `pages/MIDI-over-USB.md` and in my Doxyfile `INPUT = pages/MIDI-over-USB.md`, and `CREATE_SUBDIRS = YES`.  
Doxygen 1.8 places this file at `d8/d4a/md_pages_MIDI-over-USB.html`, while Doxygen 1.9 places it at `dc/d8a/md_pages_MIDI_over_USB.html`.  
Similarly, in Doxygen 1.8, I could refer to it using `@ref md_pages_MIDI-over-USB`, in 1.9 I have to use `@ref md_pages_MIDI_over_USB`.

Is this an intentional change? Would it be possible to restore the old behavior using a setting? There are quite a number of links that point to pages like this (some of them that I can't change), so it would be nice to preserve the current locations. (I could manually add pages that redirect to the new URLs, but I would like to avoid this.)

**Comment by albert-github** [comment]
- In which 1.8 version was it working (output of `doxygen -v`) and in which version of 1.9 ?
- it is also better to use an explicit anchor by means of the `\anchor` command or by means of the anchor possibilities of the page header (`{#...}`

This was an intentional change. The base was the issue #8156 and the pull request #8157 and optimized by means of the commit (c48c5eb3c68a9b5f3f82d1d186a6695ed1d30db5 ):
```
Commit: c48c5eb3c68a9b5f3f82d1d186a6695ed1d30db5 [c48c5eb]
Date: Thursday, November 5, 2020 7:25:15 PM
Optimized the escape function and made it more generic
```

**Comment by tttapa** [comment]
It worked in 1.8.19, and I've now tried 1.9.1.

Thanks for the tip about the explicit anchors, I didn't know it also affected the filename of the generated HTML files. I added `{#md_pages_MIDI-over-USB}` as an anchor, and it now has the same URL as before.

It might be somewhat unexpected to see dashes being replaced, since they're usually not a problem in filenames, but the anchor solution works fine for me, so I think this issue can be closed.

Thanks again!
